app/models/test_track/remote/split_registry.rb

```diff
@@ -3,35 +3,45 @@ class TestTrack::Remote::SplitRegistry
 
   CACHE_KEY = 'test_track_split_registry'.freeze
 
-  collection_path '/api/v1/split_registry'
+  collection_path '/api/v2/split_registry'
 
-  def self.fake_instance_attributes(_)
-    ::TestTrack::Fake::SplitRegistry.instance.to_h
-  end
+  class << self
+    def fake_instance_attributes(_)
+      ::TestTrack::Fake::SplitRegistry.instance.to_h
+    end
 
-  def self.instance
-    # TODO: FakeableHer needs to make this faking a feature of `get`
-    if faked?
-      new(fake_instance_attributes(nil))
-    else
-      get('/api/v1/split_registry')
+    def instance
+      # TODO: FakeableHer needs to make this faking a feature of `get`
+      if faked?
+        new(fake_instance_attributes(nil))
+      else
+        get('/api/v2/split_registry')
+      end
     end
-  end
 
-  def self.reset
-    Rails.cache.delete(CACHE_KEY)
-  end
+    def reset
+      Rails.cache.delete(CACHE_KEY)
+    end
+
+    def to_hash
+      if faked?
+        instance.attributes.freeze
+      else
+        fetch_cache { instance.attributes }.freeze
+      end
+    rescue *TestTrack::SERVER_ERRORS => e
+      Rails.logger.error "TestTrack failed to load split registry. #{e}"
+      fetch_cache { nil } # cache the missing registry for 5 seconds if we can't get one
```

Comment:
Should this return a `{}`? We recently saw a nil object error down from the `nil` return value in this method.

I'm also a fan of `to_x` methods returning an X no matter how nonsensical (and relying on input/value sanitization to handle failure states).

Reply:
for this PR there's a bunch of code that depends on this being nil, but in #83 i can have this return `{}`